File: src/ee/core/internal/Logger.swift
//
//  Logger.swift
//  Pods
//
//  Created by eps on 6/25/20.
//

import Foundation

internal class Logger: ILogger {
    private let _tag: String
    private var _logLevel: LogLevel

    init(_ tag: String, _ logLevel: LogLevel = LogLevel.Debug) {
        _tag = tag
        _logLevel = logLevel
    }

    var logLevel: LogLevel {
        get { _logLevel }
        set(value) { _logLevel = value }
    }

    func error(_ message: String) {
        if _logLevel.rawValue <= LogLevel.Error.rawValue {
            print("\(_tag): \(message)")
        }
    }

    func warn(_ message: String) {
        if _logLevel.rawValue <= LogLevel.Warn.rawValue {
            print("\(_tag): \(message)")
        }
    }

    func info(_ message: String) {
        if _logLevel.rawValue <= LogLevel.Info.rawValue {
            print("\(_tag): \(message)")
        }
    }

    func debug(_ message: String) {
        if _logLevel.rawValue <= LogLevel.Debug.rawValue {
            print("\(_tag): \(message)")
        }
    }

    func verbose(_ message: String) {
        if _logLevel.rawValue <= LogLevel.Verbose.rawValue {
            print("\(_tag): \(message)")
        }
    }
}

File: src/unity/Runtime/Core/Logger.cs
namespace EE {
    public class Logger : ILogger {
        private readonly string _tag;

        public LogLevel LogLevel { get; set; } = LogLevel.Verbose;

        public Logger(string tag) {
            _tag = tag;
        }

        public void Error(string message) {
            if (LogLevel <= LogLevel.Error) {
                UnityEngine.Debug.LogError($"{_tag}: {message}");
            }
        }

        public void Warn(string message) {
            if (LogLevel <= LogLevel.Warning) {
                UnityEngine.Debug.LogWarning($"{_tag}: {message}");
            }
        }

        public void Debug(string message) {
            if (LogLevel <= LogLevel.Warning) {
                UnityEngine.Debug.Log($"{_tag}: {message}");
            }
        }
    }
}